liblocation/src/androidTest/java/com/mapbox/android/core/location/FusedLocationEngineInstrumentedTest.java

```diff
@@ -0,0 +1,130 @@
+    when(mockLocationManager.getBestProvider(any(Criteria.class), anyBoolean()))
+      .thenReturn(LocationManager.GPS_PROVIDER);
+    doAnswer(new Answer<Object>() {
+      @Override
+      public Object answer(InvocationOnMock invocation) {
+        LocationListener listener = (LocationListener) invocation.getArguments()[3];
+        listener.onProviderEnabled(PROVIDER);
+        listener.onStatusChanged(PROVIDER, LocationProvider.AVAILABLE, null);
+        listener.onLocationChanged(location);
+        listener.onProviderDisabled(PROVIDER);
+        return null;
+      }
+    }).when(mockLocationManager)
+      .requestLocationUpdates(anyString(), anyLong(), anyFloat(), any(LocationListener.class), any(Looper.class));
+    engines.add(new LocationEngineProxy<>(new MapboxFusedLocationEngineImpl(mockContext)));
+    engines.add(new LocationEngineProxy<>(new AndroidLocationEngineImpl(mockContext)));
+  }
+
+  @Test
+  public void checkGetLastLocation() {
+    for (LocationEngineProxy engineProxy : engines) {
+      engineProxy.getLastLocation(new LocationEngineCallback<LocationEngineResult>() {
+        @Override
+        public void onSuccess(LocationEngineResult result) {
+
+        }
+
+        @Override
+        public void onFailure(@NonNull Exception exception) {
+          assertEquals("Last location unavailable", exception.getLocalizedMessage());
+        }
+      });
+
+      when(mockLocationManager.getLastKnownLocation(anyString())).thenReturn(location);
+      engineProxy.getLastLocation(new LocationEngineCallback<LocationEngineResult>() {
+        @Override
+        public void onSuccess(LocationEngineResult result) {
+          List<Location> list = result.getLocations();
+          assertEquals(1, list.size());
+          assertEquals(1.0, list.get(0).getLatitude(), 0);
+          assertEquals(2.0, list.get(0).getAltitude(), 0);
+        }
+
+        @Override
+        public void onFailure(@NonNull Exception exception) {
+
+        }
+      });
+    }
+  }
+
+  @Test
+  public void checkRequestAndRemoveLocationUpdates() {
+    LocationEngineCallback<LocationEngineResult> engineCallback = new LocationEngineCallback<LocationEngineResult>() {
+      @Override
+      public void onSuccess(LocationEngineResult result) {
+        List<Location> list = result.getLocations();
+        assertEquals(1, list.size());
+        assertEquals(1.0, list.get(0).getLatitude(), 0);
+        assertEquals(2.0, list.get(0).getAltitude(), 0);
+      }
+
+      @Override
+      public void onFailure(@NonNull Exception exception) {
+
+      }
+    };
+    for (LocationEngineProxy engineProxy : engines) {
+      engineProxy.requestLocationUpdates(getRequest(INTERVAL, LocationEngineRequest.PRIORITY_HIGH_ACCURACY),
+        engineCallback, null);
+
+      assertNotNull(engineProxy.removeListener(engineCallback));
+    }
+
+  }
+
+  private static LocationEngineRequest getRequest(long interval, int priority) {
```

Comment:
can we cover intervals, fastest intervals, max wait time with integration tests?

Reply:
These params are handled by android framework, so IMHO we don't need to test these.

Reply:
@Chaoba the location engines that we support are not in complete feature parity, hence not all configs are handled by framework. a good example is fastest interval: `MapboxFusedLocationEngine` doesn't support it: https://github.com/mapbox/mapbox-events-android/issues/312. Would be nice if we could create a test to cover all basic configs for future reference just in case someone decides to implement a new engine. Does that make sense?